### PR TITLE
Update assertTrue/False to Python 3 precise asserts

### PR DESCRIPTION
**Pull request by atombrella**
For now, I just ran some search-and-replace under the certbot directory. There are quite a lot of these; some are actually quite odd because the expression is inverted, so readability and clarity is an added bonus :) I'll do a bit more under the folder, and then ask for a review.

https://pypi.org/project/flake8-assertive/ contains an exhaustive list of replacements. I haven't run it on the project to check; one massive PR with all of the replacements would have pros, but also some con (review time). 

## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Add or update any documentation as needed to support the changes in this PR.
- [x] Include your name in `AUTHORS.md` if you like.

### CODE REVIEW

**Comment by atombrella** [comment]
I'll fix the test failures. In this work, I did find a few somewhat misleading asserts, e.g., double negated statements, `assertEqual(..., None/True/False)`. flake8-assertive still warns about some statements under certbot.

File: acme/tests/client_test.py [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ def test_urllib_error(self):
 
         # Requests Library Exceptions
         except requests.exceptions.ConnectionError as z: #pragma: no cover
-            self.assertTrue("'Connection aborted.'" in str(z) or "[WinError 10061]" in str(z))
+            self.assertIn("'Connection aborted.'" in str(z) or "[WinError 10061]", str(z))
```
Comment (alexzorin) [inline acme/tests/client_test.py:1160]: The current test failure is due to this change.

The `assertTrue` was `A in str or B in str`, so I don't think we can use a single `assertIn` here.

**Review by alexzorin — CHANGES_REQUESTED** [review]

File: acme/tests/client_test.py (hunk 1 above)
Comment (atombrella) [reply acme/tests/client_test.py]: Thanks! I used a regex to replace them, and it was obviously a bit too aggressive :/

**Review by alexzorin — CHANGES_REQUESTED** [review]
Is `self.assertIsNone()` or `self.assertIs($X, None)` preferred? Right now there is a mix. 

Here are the latter:

    $ semgrep -l py -f assert_is_none.semgrep
    running 1 rules...
    certbot/tests/cert_manager_test.py
    severity:info rule:assert_is_none: should be assertIsNone
    509:        self.assertIs(result[1], None)
    --------------------------------------------------------------------------------
    514:        self.assertIs(result[0], None)

    certbot/tests/cli_test.py
    severity:info rule:assert_is_none: should be assertIsNone
    428:        self.assertIs(namespace.deploy_hook, None)
    --------------------------------------------------------------------------------
    467:        self.assertIs(namespace.delete_after_revoke, None)

    certbot/tests/helpful_test.py
    severity:info rule:assert_is_none: should be assertIsNone
    93:        self.assertIs(parsed_args.eab_kid, None)

    certbot/tests/plugins/selection_test.py
    severity:info rule:assert_is_none: should be assertIsNone
    183:        self.assertIs(self._call(), None)

    certbot/tests/storage_test.py
    severity:info rule:assert_is_none: should be assertIsNone
    285:        self.assertIs(self.test_rc.current_version("cert"), None)
    --------------------------------------------------------------------------------
    770:        self.assertIs(self.test_rc.server, None)
    ran 1 rules on 364 files: 8 findings

**Comment by atombrella** [comment]
> Is `self.assertIsNone()` or `self.assertIs($X, None)` preferred? Right now there is a mix.

I think `assertIsNone`. I'll update. Most of these replacements were done with a regular expressions.

Thanks for pointing out `semgrep`, it seems quite useful! I'll try to work with this and also `flake8-assertive` to get the last ones cleared out under the `acme` and `certbot` directories. It's too much work and review effort from you to review the replacements done throughout the entire repository, so chunk/component by chunk is better.

**Comment by atombrella** [comment]
@alexzorin There's one left that `flake8-assertive` complains about, however, it's not possible to replace it without touching `certbot.plugins.dns_test_common._AuthenticatorCallableTestCase`.

**Review by alexzorin — APPROVED** [review]
Great. I think it's to pick up the rest later if/once we get flake8/black rolling.

**Pull request merged**